src/net/server/handlers/create_query.rs

```diff
@@ -0,0 +1,246 @@
+        match field_type.as_str() {
+            ff::Fp2::STR_REPR => Ok(ByteArrStream::new(body, ff::Fp2::SIZE_IN_BYTES)),
+            ff::Fp31::STR_REPR => Ok(ByteArrStream::new(body, ff::Fp31::SIZE_IN_BYTES)),
+            ff::Fp32BitPrime::STR_REPR => {
+                Ok(ByteArrStream::new(body, ff::Fp32BitPrime::SIZE_IN_BYTES))
+            }
+            other => Err(MpcHelperServerError::bad_query_value(&field_type, other)),
+        }
+    }
+}
+
+#[allow(clippy::unused_async)]
+pub async fn handler(_body: ByteArrStream) -> Result<(), MpcHelperServerError> {
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use axum::http::Request;
+    use futures_util::{StreamExt, TryStreamExt};
+    use hyper::Body;
+    use rand::distributions::Uniform;
+    use rand::{thread_rng, Rng};
+    use rand_core::{CryptoRng, RngCore};
+
+    fn random_vec<R: RngCore + CryptoRng>(rng: &mut R, len: usize) -> Vec<u8> {
+        let range = Uniform::from(0..u8::MAX);
+        rng.sample_iter(&range).take(len).collect()
+    }
+
+    async fn to_byte_arr_stream(
+        slice: &[u8],
+        field_type: &'static str,
+    ) -> Result<ByteArrStream, MpcHelperServerError> {
+        let b = Body::from(Bytes::from(slice.to_owned()));
+        let mut req_parts = RequestParts::new(
+            Request::post(format!("/example?field_type={field_type}"))
+                .body(b)
+                .unwrap(),
+        );
+        ByteArrStream::from_request(&mut req_parts).await
+    }
+
+    #[tokio::test]
+    async fn byte_arr_stream_produces_bytes_fp2() {
+        let vec = random_vec(&mut thread_rng(), 10);
+        let stream = to_byte_arr_stream(&vec, ff::Fp2::STR_REPR).await.unwrap();
+        let collected = stream.try_collect::<Vec<_>>().await.unwrap();
+        for (expected, got) in vec.chunks(ff::Fp2::SIZE_IN_BYTES as usize).zip(collected) {
+            assert_eq!(expected, got.as_ref());
+        }
+    }
+
+    #[tokio::test]
+    async fn byte_arr_stream_produces_bytes_fp32_bit_prime() {
+        const ARR_SIZE: usize = 20;
+        let vec = random_vec(&mut thread_rng(), ARR_SIZE * 10);
+        let stream = to_byte_arr_stream(&vec, ff::Fp32BitPrime::STR_REPR)
+            .await
+            .unwrap();
+        let collected = stream.try_collect::<Vec<_>>().await.unwrap();
+        for (expected, got) in vec
+            .chunks(ff::Fp32BitPrime::SIZE_IN_BYTES as usize)
+            .zip(collected)
+        {
+            assert_eq!(expected, got.as_ref());
+        }
+    }
+
+    #[tokio::test]
+    async fn byte_arr_stream_fails_with_bad_field_type() {
+        let vec = random_vec(&mut thread_rng(), 8);
+        let stream = to_byte_arr_stream(&vec, "bad_field_type").await;
+        assert!(matches!(
+            stream,
+            Err(MpcHelperServerError::BadQueryString(_))
+        ));
+    }
+
+    #[tokio::test]
+    async fn byte_arr_stream_fails_on_invalid_size() {
```

Comment:
Would it be possible to write a property test that picks up a body that is sliced into different size chunks (but total size is aligned with field size) and makes sure it produces correct output?

Reply:
done